src/main/java/core/basesyntax/service/operation/SupplyOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.Storage;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Storage storage) {
```

Comment:
Please ensure that the `handle` method checks the result balance before saving it in the Storage according to the checklist item 'Handling Purchase operation'. Even though this is a supply operation, it is a good practice to validate the quantity to ensure it's not negative.